applications/minotari_ledger_wallet/wallet/src/main.rs

```diff
@@ -40,62 +36,27 @@ use handlers::{
     get_version::handler_get_version,
     get_view_key::handler_get_view_key,
 };
+#[cfg(not(any(target_os = "stax", target_os = "flex")))]
+use ledger_device_sdk::io::Event;
+use ledger_device_sdk::io::{ApduHeader, Comm, Reply, StatusWords};
+#[cfg(any(target_os = "stax", target_os = "flex"))]
+use ledger_device_sdk::nbgl::{init_comm, NbglHomeAndSettings, StatusType};
 #[cfg(feature = "pending_review_screen")]
 use ledger_device_sdk::ui::gadgets::display_pending_review;
-use ledger_device_sdk::{
-    io::{ApduHeader, Comm, Event, Reply, StatusWords},
-    ui::gadgets::SingleMessage,
-};
 use minotari_ledger_wallet_common::common_types::{
     AppSW as AppSWMapping,
     Branch as BranchMapping,
     Instruction as InstructionMapping,
 };
-
 ledger_device_sdk::set_panic!(ledger_device_sdk::exiting_panic);
 
 static BIP32_COIN_TYPE: u32 = 535348;
 static CLA: u8 = 0x80;
 static RESPONSE_VERSION: u8 = 1;
 
-/// Allocator heap size
-const HEAP_SIZE: usize = 1024 * 26;
-
-/// Statically allocated heap memory
-static mut HEAP_MEM: [MaybeUninit<u8>; HEAP_SIZE] = [MaybeUninit::uninit(); HEAP_SIZE];
-
-/// Bind global allocator
-#[global_allocator]
-static HEAP: embedded_alloc::Heap = embedded_alloc::Heap::empty();
-
-/// Error handler for allocation
-#[alloc_error_handler]
-fn alloc_error(_: core::alloc::Layout) -> ! {
-    SingleMessage::new("allocation error!").show_and_wait();
-    ledger_device_sdk::exit_app(250)
-}
-
-/// Initialise allocator
-pub fn init() {
-    unsafe { HEAP.init(HEAP_MEM.as_ptr() as usize, HEAP_SIZE) }
-}
-
-struct MyCriticalSection;
-critical_section::set_impl!(MyCriticalSection);
-
-unsafe impl critical_section::Impl for MyCriticalSection {
-    unsafe fn acquire() -> RawRestoreState {
-        // nothing, it's all good, don't worry bout it
-    }
-
-    unsafe fn release(_token: RawRestoreState) {
-        // nothing, it's all good, don't worry bout it
-    }
-}
```

Comment:
Was this tested on the nanos+ ? 

I was under the distinct impression without defining the allocator we lose the ability to well... Allocate anything. Essentially it's memory-less

Reply:
yes it was, 
Ledger realised that they need one, so the rust_sdk lib now includes one and it clashes as you can only have one defined.

Reply:
Ahhh neat.